src/common/redis_module/ray_redis_module.c

```diff
@@ -272,6 +283,213 @@ int ResultTableLookup_RedisCommand(RedisModuleCtx *ctx,
+  if (task_spec == NULL) {
+    RedisModule_HashSet(key, REDISMODULE_HASH_CFIELDS, "state", state, "node",
+                        node_id, NULL);
+    RedisModule_HashGet(key, REDISMODULE_HASH_CFIELDS, "task_spec", &task_spec,
+                        NULL);
+    if (task_spec == NULL) {
+      return RedisModule_ReplyWithError(
+          ctx, "Cannot update a task that doesn't exist yet");
+    }
+  } else {
+    RedisModule_HashSet(key, REDISMODULE_HASH_CFIELDS, "state", state, "node",
+                        node_id, "task_spec", task_spec, NULL);
+  }
+  RedisModule_CloseKey(key);
+
+  /* Build the PUBLISH topic and message for task table subscribers. The topic
+   * is a string in the format "TASK_PREFIX:<state>:<node ID>". The
+   * message is a string in the format: "<task ID> <state> <node ID> <task
+   * specification>". */
+  const char *publish_field;
+  RedisModuleString *publish_topic =
+      RedisModule_CreateStringPrintf(ctx, "%s", TASK_PREFIX);
+  RedisModuleString *publish_message =
+      RedisModule_CreateStringFromString(ctx, task_id);
+  /* Append the scheduling state. */
+  publish_field = RedisModule_StringPtrLen(state, &length);
+  RedisModule_StringAppendBuffer(ctx, publish_topic, publish_field, length);
+  RedisModule_StringAppendBuffer(ctx, publish_message, " ", strlen(" "));
+  RedisModule_StringAppendBuffer(ctx, publish_message, publish_field, length);
+  /* Append the node ID. */
+  publish_field = RedisModule_StringPtrLen(node_id, &length);
+  RedisModule_StringAppendBuffer(ctx, publish_topic, ":", strlen(":"));
+  RedisModule_StringAppendBuffer(ctx, publish_topic, publish_field, length);
+  RedisModule_StringAppendBuffer(ctx, publish_message, " ", strlen(" "));
+  RedisModule_StringAppendBuffer(ctx, publish_message, publish_field, length);
+  /* Append the task specification. */
+  publish_field = RedisModule_StringPtrLen(task_spec, &length);
+  RedisModule_StringAppendBuffer(ctx, publish_message, " ", strlen(" "));
+  RedisModule_StringAppendBuffer(ctx, publish_message, publish_field, length);
+
+  RedisModuleCallReply *reply =
+      RedisModule_Call(ctx, "PUBLISH", "ss", publish_topic, publish_message);
+  if (reply == NULL) {
+    return RedisModule_ReplyWithError(ctx, "PUBLISH unsuccessful");
+  }
+
+  RedisModule_FreeString(ctx, publish_message);
+  RedisModule_FreeString(ctx, publish_topic);
+  RedisModule_ReplyWithSimpleString(ctx, "ok");
+
+  return REDISMODULE_OK;
+}
+
+/**
+ * Add a new entry to the task table. This will overwrite any existing entry
+ * with the same task ID.
+ *
+ * This is called from a client with the command:
+ *
+ *     RAY.task_table_add <task ID> <state> <node ID> <task spec>
+ *
+ * @param task_id A string that is the ID of the task.
+ * @param state A string that is the current scheduling state (a
+ *        scheduling_state enum instance). The string's value must be an
+ *        integer with width less than or equal to 2. If the width is less than
```

Comment:
Can we add a check for this condition?

Also, in the documentation, can you phrase it more like "the string must be a nonnegative integer less than 100 (so that it can be left-padded with spaces to length 2)"